pydle/features/rfc1459/client.py

```diff
@@ -59,6 +59,24 @@ def _reset_attributes(self):
         self.channels = parsing.NormalizingDict(self.channels, case_mapping=self._case_mapping)
         self.users = parsing.NormalizingDict(self.users, case_mapping=self._case_mapping)
 
+        # List.
+        self.all_channels = []
+        self._list_client = []
+        self._list_channel = []
+        self._list_count = []
+        self._list_topic = []
+        self._list_query = asyncio.Queue()
+
+    async def channel_list(self):
+        await self.rawmsg("LIST")
+        future = asyncio.get_running_loop().create_future()
+        await self._list_query.put(future)
```

Comment:
This future should be enqueued prior to emitting the `LIST` command, to prevent a race.